File: alipay/src/main/java/com/king/pay/alipay/AliPayReq.java
package com.king.pay.alipay;

/**
 * 参数说明请参见：<a href="https://docs.open.alipay.com/204/105465/">支付宝支付请求参数说明</a>
 * @author <a href="mailto:jenly1314@gmail.com">Jenly</a>
 */
public class AliPayReq {

    /** 支付宝支付分配的应用ID */
    private String appId;
    /** 接口名称 示例值：alipay.trade.app.pay */
    private String method = "alipay.trade.app.pay";
    /** 格式 */
    private String format = "JSON";
    /** 请求使用的编码格式，如utf-8,gbk,gb2312等 */
    private String charset = "utf-8";
    /** 签名类型 示例值：RSA2 */
    private String signType = "RSA2";
    /** 签名 */
    private String sign;
    /** 时间戳 */
    private String timestamp;
    /** 调用的接口版本，固定为：1.0 */
    private String version = "1.0";
    /** 支付宝服务器主动通知商户服务器里指定的页面http/https路径。建议商户使用https */
    private String notifyUrl;
    /** 业务请求参数的集合 */
    private String bizContent;

    public String getAppId() {
        return appId;
    }

    public AliPayReq setAppId(String appId) {
        this.appId = appId;
        return this;
    }

    public String getMethod() {
        return method;
    }

    public AliPayReq setMethod(String method) {
        this.method = method;
        return this;
    }

    public String getFormat() {
        return format;
    }

    public AliPayReq setFormat(String format) {
        this.format = format;
        return this;
    }

    public String getCharset() {
        return charset;
    }

    public AliPayReq setCharset(String charset) {
        this.charset = charset;
        return this;
    }

    public String getSignType() {
        return signType;
    }

    public AliPayReq setSignType(String signType) {
        this.signType = signType;
        return this;
    }

    public String getSign() {
        return sign;
    }

    public AliPayReq setSign(String sign) {
        this.sign = sign;
        return this;
    }

    public String getTimestamp() {
        return timestamp;
    }

    public AliPayReq setTimestamp(String timestamp) {
        this.timestamp = timestamp;
        return this;
    }

    public String getVersion() {
        return version;
    }

    public AliPayReq setVersion(String version) {
        this.version = version;
        return this;
    }

    public String getNotifyUrl() {
        return notifyUrl;
    }

    public AliPayReq setNotifyUrl(String notifyUrl) {
        this.notifyUrl = notifyUrl;
        return this;
    }

    public String getBizContent() {
        return bizContent;
    }

    public AliPayReq setBizContent(String bizContent) {
        this.bizContent = bizContent;
        return this;
    }
}

File: alipay/src/main/java/com/king/pay/alipay/AliPayReq.kt
package com.king.pay.alipay

/**
 * 参数说明请参见：<a href="https://docs.open.alipay.com/204/105465/">支付宝支付请求参数说明</a>
 *
 * @author <a href="mailto:jenly1314@gmail.com">Jenly</a>
 * <p>
 * <a href="https://github.com/jenly1314">Follow me</a>
 */
@Suppress("unused")
open class AliPayReq {

    /**
     * 支付宝支付分配的应用ID
     */
    var appId: String? = null
        private set

    /**
     * 接口名称 示例值：alipay.trade.app.pay
     */
    var method: String = "alipay.trade.app.pay"
        private set

    /**
     * 格式
     */
    var format: String = "JSON"
        private set

    /**
     * 请求使用的编码格式，如utf-8,gbk,gb2312等
     */
    var charset: String = "utf-8"
        private set

    /**
     * 签名类型 示例值：RSA2
     */
    var signType: String = "RSA2"
        private set

    /**
     * 签名
     */
    var sign: String? = null
        private set

    /**
     * 时间戳
     */
    var timestamp: String? = null
        private set

    /**
     * 调用的接口版本，固定为：1.0
     */
    var version: String = "1.0"
        private set

    /**
     * 支付宝服务器主动通知商户服务器里指定的页面http/https路径。建议商户使用https
     */
    var notifyUrl: String? = null
        private set

    /**
     * 业务请求参数的集合
     */
    var bizContent: String? = null
        private set

    fun getAppId(): String? = appId

    fun setAppId(appId: String?): AliPayReq = apply { this.appId = appId }

    fun getMethod(): String = method

    fun setMethod(method: String): AliPayReq = apply { this.method = method }

    fun getFormat(): String = format

    fun setFormat(format: String): AliPayReq = apply { this.format = format }

    fun getCharset(): String = charset

    fun setCharset(charset: String): AliPayReq = apply { this.charset = charset }

    fun getSignType(): String = signType

    fun setSignType(signType: String): AliPayReq = apply { this.signType = signType }

    fun getSign(): String? = sign

    fun setSign(sign: String?): AliPayReq = apply { this.sign = sign }

    fun getTimestamp(): String? = timestamp

    fun setTimestamp(timestamp: String?): AliPayReq = apply { this.timestamp = timestamp }

    fun getVersion(): String = version

    fun setVersion(version: String): AliPayReq = apply { this.version = version }

    fun getNotifyUrl(): String? = notifyUrl

    fun setNotifyUrl(notifyUrl: String?): AliPayReq = apply { this.notifyUrl = notifyUrl }

    fun getBizContent(): String? = bizContent

    fun setBizContent(bizContent: String?): AliPayReq = apply { this.bizContent = bizContent }
}
